rskj-core/src/test/java/co/rsk/peg/BridgeSupportPegoutTransactionCreatedEventTest.java

```diff
@@ -0,0 +1,323 @@
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import co.rsk.bitcoinj.core.BtcTransaction;
+import co.rsk.bitcoinj.core.Coin;
+import co.rsk.bitcoinj.core.Context;
+import co.rsk.bitcoinj.core.NetworkParameters;
+import co.rsk.bitcoinj.core.Sha256Hash;
+import co.rsk.bitcoinj.core.TransactionOutput;
+import co.rsk.bitcoinj.core.UTXO;
+import co.rsk.bitcoinj.wallet.Wallet;
+import co.rsk.blockchain.utils.BlockGenerator;
+import co.rsk.config.BridgeConstants;
+import co.rsk.config.BridgeMainNetConstants;
+import co.rsk.crypto.Keccak256;
+import co.rsk.peg.ReleaseRequestQueue.Entry;
+import co.rsk.peg.federation.Federation;
+import co.rsk.peg.federation.FederationArgs;
+import co.rsk.peg.federation.FederationFactory;
+import co.rsk.peg.federation.FederationTestUtils;
+import co.rsk.peg.utils.BridgeEventLogger;
+import co.rsk.test.builders.BridgeSupportBuilder;
+import java.io.IOException;
+import java.time.Instant;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Optional;
+import java.util.TreeMap;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import org.ethereum.config.blockchain.upgrades.ActivationConfig;
+import org.ethereum.config.blockchain.upgrades.ActivationConfigsForTest;
+import org.ethereum.config.blockchain.upgrades.ConsensusRule;
+import org.ethereum.core.Transaction;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+class BridgeSupportPegoutTransactionCreatedEventTest {
+
+    private static final BridgeConstants bridgeMainnetConstants = BridgeMainNetConstants.getInstance();
+    private static final NetworkParameters btcMainnetParams = bridgeMainnetConstants.getBtcParams();
+
+    private BridgeStorageProvider provider;
+    private BridgeEventLogger eventLogger;
+
+    @BeforeEach
+    void init() throws IOException {
+        provider = mock(BridgeStorageProvider.class);
+        eventLogger = mock(BridgeEventLogger.class);
+
+        when(provider.getFeePerKb())
+            .thenReturn(Coin.MILLICOIN);
+
+        when(provider.getPegoutsWaitingForSignatures())
+            .thenReturn(new TreeMap<>());
+
+        when(provider.getPegoutsWaitingForConfirmations())
+            .thenReturn(new PegoutsWaitingForConfirmations(new HashSet<>()));
+    }
+
+    private static Stream<Arguments> pegoutTransactionCreatedEventArgsProvider() {
+        return Stream.of(
+            Arguments.of(ActivationConfigsForTest.arrowhead600().forBlock(0)),
+            Arguments.of(ActivationConfigsForTest.lovell700().forBlock(0))
+        );
+    }
+
+    @ParameterizedTest
+    @MethodSource("pegoutTransactionCreatedEventArgsProvider")
+    void test_pegoutTransactionCreatedEvent_when_pegout_batch_is_created(ActivationConfig.ForBlock activations) throws IOException {
```

Comment:
`pegoutTransactionCreatedEvent_whenPegoutBatchCreation_ shouldDoSomething`
something like that, wdyt?

Reply:
Done